contracts/amm/SirenExchange.sol

```diff
@@ -0,0 +1,308 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity >=0.5.0 <=0.8.0;
+import "../amm/ISirenTradeAMM.sol";
+import "../series/SeriesLibrary.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+import "@uniswap/v2-periphery/contracts/interfaces/IUniswapV2Router02.sol";
+import "@openzeppelin/contracts/token/ERC1155/utils/ERC1155Holder.sol";
+import "@uniswap/lib/contracts/libraries/TransferHelper.sol";
+
+/// This is an implementation of exchanging collateral tokens for different collateral tokens that represent the underlying asset for series in SirenMarkets
```

Comment:
"This is an implementation of exchanging collateral tokens for different collateral tokens"

I think we should use a different term for the token that the user owns. In the tests we use the term "payment token" which works for bTokenBuy, since they're paying with that token, but it's a little clunky for the bTokenSell/wTokenSell functions where they don't pay with that token, but rather *receive* that token.

Could we use `user token` or `external token`? If you come up with something you like more let's use that. But I think right now it's confusing more than helping to use the term "collateral token" for the token the user owns, because it really isn't collateral.

Also in this line, and throughout the contract's comments, we shouldn't make any mention of "underlying token". It is not relevant to what's going on in the SirenExchange. Only the collateral token and the "user token" (whatever we decide to call it) is important. I think maybe we confused "underlying" and "collateral".